src/main/java/io/anserini/collection/DocumentCollection.java

```diff
@@ -88,19 +93,72 @@ public final Path getCollectionPath() {
     return path;
   }
 
+  /**
+   * Creates a {@code FileSegment} from a path.
+   *
+   * @param p path
+   * @return {@code FileSegment} with the specified path
+   * @throws IOException if file access error encountered
+   */
+  public abstract FileSegment<T> createFileSegment(Path p) throws IOException;
```

Comment:
Do we still need this? Or shall we let the iterator figure out how to this itself?